### ISSUE TITLE
[Question] Train/Validation/Test Splits

### ISSUE THREAD

**Issue by nathanmlim** [chemprop/chemprop#58]
When I provide a single CSV file containing **all** my molecules to `chemprop_train --data_path` and chemprop does the train/validation/test split, does chemprop save the identities which belong to the validation/test sets such that I can use the same file for `chemprop_predict --test_path` ? Specifically, I'm wondering if I have to remove molecules which appeared in the training dataset from my CSV file before passing it to `chemprop_predict --test_path`

**Comment by swansonk14** [comment]
Hi @nathanmlim, that's a good question. When you run `chemprop_predict`, you will need to specify the test molecules. If you simply set `--test_path` to be the same as `--data_path`, then it will make predictions on all the molecules, not just the test molecules. The easiest way to separate out the test set is to add `--save_smiles_splits --save_dir ckpt` when running `chemrop_train`. This will save each of the train, validation, and test split smiles in the directory `ckpt` along with the models. Then when you want to make predictions, you can set `--test_path ckpt/fold_0/test_smiles.csv` to make predictions just on the test set.

**Comment by nathanmlim** [comment]
Hi @swansonk14, thanks for you reply. I added that flag during training and only see `train_full.csv` and a `train_smiles.csv` file in the `fold_0/` path. Any ideas as to why it did not save the test/validation set?

**Comment by nathanmlim** [comment]
Upon closer inspection it looks like my processes are being killed for one reason or another (will post a separate issue) so this may explain why it is not saving the test/validation sets.
